@here/harp-mapview/lib/Utils.ts

```diff
@@ -367,17 +366,14 @@ export namespace MapViewUtils {
         position: THREE.Vector3,
         projection: geoUtils.Projection
     ): THREE.Vector3 {
-        // Position on the ground for surface normal calculation does not matter for
-        // any planar projections, so we may simplify calculus by returning const vector:
-        if (projection.type === geoUtils.ProjectionType.Planar) {
-            return groundNormalInvPlanarProj;
-        } else {
-            // Note using different vector then caller, otherwise vector data could be
-            // silently modified.
-            const normal: THREE.Vector3 = tmpVectors[1];
-            projection.surfaceNormal(position, normal);
-            return normal.negate();
-        }
+        // We can't simply return [0,0,1] for planar projections because WebMercatorProjection
+        // has ground normal inverted - as opposite to other planar projections, so use of
+        // the projection interface is encouraged to acquire ground normal.
+        // Note using different (tmp) vector then caller, otherwise vector data could be
+        // silently modified.
+        const normal: THREE.Vector3 = tmpVectors[1];
```

Comment:
I know it's not your code, but I would change the interface of this method to:
```
    function getGroundNormalInv(
        position: THREE.Vector3,	        position: THREE.Vector3,
        projection: geoUtils.Projection	        projection: geoUtils.Projection
        result?: THREE.Vector3
    ): THREE.Vector3
```
Otherwise the behavior of this method is quite confusing:
```
const normal1 = getGroundNormalInv(position1, projection);
const normal2 = getGroundNormalInv(position2, projection);
```
If result is undefined this method would allocate a new Vector3. It would be basically up to the user whether an allocation happens or not.

Reply:
Will fix it too.